linera-base/src/crypto/secp256k1.rs

```diff
@@ -34,17 +45,33 @@ pub struct Secp256k1KeyPair {
 
 /// A secp256k1 signature.
 #[derive(Eq, PartialEq, Copy, Clone)]
-pub struct Secp256k1Signature(pub secp256k1::ecdsa::Signature);
+pub struct Secp256k1Signature(pub Signature);
 
 impl Secp256k1PublicKey {
     /// A fake public key used for testing.
     #[cfg(with_testing)]
     pub fn test_key(seed: u8) -> Self {
         use rand::SeedableRng;
-
         let mut rng = rand::rngs::StdRng::seed_from_u64(seed as u64);
-        let secp = secp256k1::Secp256k1::signing_only();
-        Self(secp.generate_keypair(&mut rng).1)
+        let sk = k256::SecretKey::random(&mut rng);
+        Self(sk.public_key().into())
+    }
+
+    /// Returns the bytes of the public key in compressed representation.
+    pub fn as_bytes(&self) -> Vec<u8> {
+        self.0.to_encoded_point(true).as_bytes().to_vec()
+    }
+
+    /// Decodes the bytes into the public key.
+    /// Expects the bytes to be of compressed representation.
+    ///
+    /// Panics if the encoding can't be done in a constant time.
+    pub fn from_bytes(bytes: &[u8]) -> Self {
+        Self(
+            k256::PublicKey::from_encoded_point(&EncodedPoint::from_bytes(bytes).unwrap())
+                .expect("Decode in constant time.")
+                .into(),
```

Comment:
TODO @deuszx : fix this.

Reply:
If this TODO (and others) isn't to be fixed in the PR, can we get an issue + comment in the code?

Reply:
I'm working on these right now. Apologies for confusion.